open-metadata-implementation/admin-services/admin-services-server/src/main/java/org/odpi/openmetadata/adminservices/OMAGServerAdminForViewServices.java

```diff
@@ -313,11 +315,44 @@ public VoidResponse configureAllViewServices(String userId,
     private ViewServiceConfig createViewServiceConfig(ViewServiceRegistration     registration,
                                                       ViewServiceConfig           requestedViewServiceConfig)
     {
-        ViewServiceConfig viewServiceConfig = new ViewServiceConfig(registration);
+
+        ViewServiceConfig viewServiceConfig = null;
+
+        if (requestedViewServiceConfig instanceof IntegrationViewServiceConfig) {
+            /*
+             * The requested configuration is for an Integration View Service
+             */
+            IntegrationViewServiceConfig requestedIntegrationViewServiceConfig = (IntegrationViewServiceConfig)requestedViewServiceConfig;
+            IntegrationViewServiceConfig createdViewServiceConfig = new IntegrationViewServiceConfig(registration);
+            createdViewServiceConfig.setResourceEndpoints(requestedIntegrationViewServiceConfig.getResourceEndpoints());
+            viewServiceConfig = (ViewServiceConfig) createdViewServiceConfig;
+        }
+        else if (requestedViewServiceConfig instanceof SolutionViewServiceConfig) {
+            /*
+             * The requested configuration is for a Solution View Service
+             */
+            SolutionViewServiceConfig requestedSolutionViewServiceConfig = (SolutionViewServiceConfig)requestedViewServiceConfig;
+            SolutionViewServiceConfig createdViewServiceConfig = new SolutionViewServiceConfig(registration);
+            createdViewServiceConfig.setOMAGServerPlatformRootURL(requestedSolutionViewServiceConfig.getOMAGServerPlatformRootURL());
```

Comment:
this line and next line could be moved out of the ifs -as these fields exist in ViewServiceConfig. They can be removed from the integration service code as well.

Reply:
Do you mean the omagServerName ad omagServerPlatformRootURL? They are only relevant on Solution and Vanilla configs; they are bot used at all in an Integration view service